python/examples/example_errors.ipynb

```diff
@@ -0,0 +1,991 @@
+   "source": [
+    "**How to address?**\n",
+    "\n",
+    "The error message already suggests a fix for this situation, so let's try increasing the relative tolerance:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3d1552e3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "amici_solver = amici_model.getSolver()\n",
+    "amici_solver.setRelativeTolerance(200 * amici_solver.getRelativeTolerance())\n",
+    "\n",
+    "np.random.seed(4)\n",
+    "problem_parameters = dict(\n",
+    "    zip(\n",
+    "        petab_problem.x_free_ids,\n",
+    "        petab_problem.sample_parameter_startpoints(n_starts=1)[0],\n",
+    "    )\n",
+    ")\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True,\n",
+    "    solver=amici_solver\n",
+    ")\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])\n",
+    "assert all(rdata.status == amici.AMICI_SUCCESS for rdata in res[RDATAS])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "86f4db29",
+   "metadata": {},
+   "source": [
+    "## `AMICI encountered a NaN / Inf value for [...]`\n",
+    "\n",
+    "Let's run a simulation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d97349ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "petab_problem = benchmark_models_petab.get_problem(\"Borghans_BiophysChem1997\")\n",
+    "amici_model = import_petab_problem(petab_problem, verbose=False)\n",
+    "\n",
+    "np.random.seed(18)\n",
+    "problem_parameters = dict(\n",
+    "    zip(\n",
+    "        petab_problem.x_free_ids,\n",
+    "        petab_problem.sample_parameter_startpoints(n_starts=1)[0],\n",
+    "    )\n",
+    ")\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True\n",
+    ")\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])\n",
+    "assert [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]] == ['AMICI_FIRST_RHSFUNC_ERR']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "63641cff",
+   "metadata": {},
+   "source": [
+    "**What happened?**\n",
+    "\n",
+    "The forward simulation failed because AMICI encountered a `NaN` value when simulating condition `model1_data1`.\n",
+    "Then `NaN`s occurred in $\\dot x$ and $w$ (model expressions, such as reaction fluxes or assignment rules). Furthermore, the failure occurred at the first call, so at $t = t_0$ (here: $t = 0$).\n",
+    "\n",
+    "**How to address?**\n",
+    "\n",
+    "The `NaN` in $\\dot x$ is most likely a consequence of the one in $w$. (A subset of) the dependency tree looks something like:\n",
```

Comment:
J, xdot and y also all depend on expressions w

Reply:
Thanks. And a couple more connections were missing.